robot/EDA/resources/EDA.py

```diff
@@ -381,3 +424,187 @@ def get_namespace_prefix(self, name):
+        element_menu = self.selenium.driver.find_element_by_xpath(locator_menu)
+        locator_tab = eda_lex_locators["navigation_tab"].format(tab)
+
+        self.selenium.wait_until_page_contains_element(
+            locator_menu,
+            error="Navigation menu drop down unavailable"
+        )
+        # javascript is being used here because the usual selenium click is highly unstable for this element on MetaCI
+        self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+        time.sleep(1)
+
+        # Sometimes, single click fails. Hence an additional condition to click on it again
+        if not self.check_if_element_exists(locator_tab):
+            self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+            time.sleep(1)
+
+        self.selenium.capture_page_screenshot()
+        self.selenium.wait_until_page_contains_element(
+            locator_tab,
+            error=tab + " item not found as an available option in: " + locator_tab
+        )
+        self.selenium.click_element(locator_tab)
+
+
+
+    def select_tab(self, tab):
+        """ Selects Contacts tab - as passed in to the function """
+        locator_menu = eda_lex_locators["tab_menu"].format(tab)
+        element_menu = self.selenium.driver.find_element_by_xpath(locator_menu)
+        locator_tab = eda_lex_locators["tab_tab"].format(tab)
+        self.selenium.wait_until_page_contains_element(
+            locator_menu,
+            error="Contact tab unavailable"
+        )
+        # javascript is being used here because the usual selenium click is highly unstable for this element on MetaCI
+        self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+        time.sleep(1)
+
+        # Sometimes, single click fails. Hence an additional condition to click on it again
+        if not self.check_if_element_exists(locator_tab):
+            self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+            time.sleep(1)
+
+
+    def select_value_from_listbox(self, title, value):
+        """ Selects value from a listbox/dropdown identified by title """
+        locator_title = eda_lex_locators["listbox"]["title"].format(title)
+        locator_value = eda_lex_locators["listbox"]["value"].format(value)
+        self.selenium.click_element(locator_title)
+        time.sleep(1)
+        self.selenium.click_element(locator_value)
+
+    def select_xpath(self, loc, value):
+        """ Selects the correct xpath by checking if it exists on the page
+            from the given list of locator possibilities
+        """
+        locators = eda_lex_locators[loc].values()
+
+        for i in locators:
+            locator = self.format_all(i, value)
+            if self.check_if_element_exists(locator):
+                return locator
+
+        assert "Button with the provided locator not found"
+
+    def wait_for_locator_with_timeout(self, timeout, path, *args):
```

Comment:
Why did you write this keyword, instead of using one of the built-in keywords such as [Wait until page contains element](http://robotframework.org/SeleniumLibrary/SeleniumLibrary.html#Wait%20Until%20Page%20Contains%20Element)?  I recommend using the built-in keyword unless there's a compelling reason not to. If there is a compelling reason, it needs to be spelled out in the documentation for this keyword.

Reply:
@boakley - this isn't my code.  I'm not sure who wrote it.  And I don't know when it was inserted into this file.

Reply:
@dashattuck For all of the code that was written by somebody else - why would they be showing up as new additions now? (Or is this code being copied over from some other project, like SAL?)

Reply:
@boakley - agreed.  I'm not sure why they wrote it this way either.  I'll see if I can modify their code to use it correctly.

Reply:
I don't understand why you marked this conversation as resolved. It doesn't look like you changed anything.

Reply:
@boakley - I won't mark the conversations resolved - misunderstanding on my part.

Reply:
I removed this function.